### PR TITLE
pci_hotplug: cancel test if pci not given

### PR DESCRIPTION
**Pull request by abdhaleegit**
make sure the default value is porper and skip the test
for all non pci devices

Signed-off-by: Abdul Haleem <abdhalee@linux.vnet.ibm.com>

### CODE REVIEW

File: io/pci/pci_hotplug.py [hunk 1]
```diff
@@ -58,10 +58,12 @@ def setUp(self):
             if not linux_modules.module_is_loaded("pnv_php"):
                 linux_modules.load_module("pnv_php")
         self.dic = {}
-        self.device = self.params.get('pci_devices', default=' ').split(" ")
-        self.count = int(self.params.get('count', default='1'))
-        if not self.device:
+        self.device = self.params.get('pci_devices', default=None)
+        if self.device:
+            self.device = self.device.split(" ")
+        else:
```
Comment (narasimhan-v) [inline io/pci/pci_hotplug.py:64]:
```
if not self.device:
    self.cancel("PCI_address not given")
self.device = self.device.split(" ")
```

File: io/pci/PowerVMEEH.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def setUp(self):
         if '0x1' not in genio.read_file(eeh_enable_file).strip():
             self.cancel("EEH is not enabled, please enable via FSP")
         self.max_freeze = self.params.get('max_freeze', default=1)
-        self.pci_addr = [self.params.get('pci_device', default='')]
+        self.pci_addr = [self.params.get('pci_device', default=None)]
```
Comment (narasimhan-v) [inline io/pci/PowerVMEEH.py:69]: Please have this as a separate PR.
We will need to review Readme file to see if space is what it is separated by.

File: io/pci/pci_hotplug.py (hunk 1 above)
Comment (Naresh-ibm) [inline io/pci/pci_hotplug.py:64]: the existing logic is taken care of cancelling if pci_devices are not given, then may I know why we are changing this ?

**Comment by narasimhan-v** [comment]
@abdhaleegit Is this still being pursued ?

File: io/pci/pci_hotplug.py (hunk 1 above)
Comment (abdhaleegit) [reply io/pci/pci_hotplug.py]: self.device = self.params.get('pci_devices', default=' ').split(" ")  >> was failing with error if no pci device was given.

so first check if pci present and than split is the right way

**Comment by abdhaleegit** [comment]
@vnarasimhan included all the comments, and this can be merged

File: io/pci/PowerVMEEH.py (hunk 2 above)
Comment (abdhaleegit) [reply io/pci/PowerVMEEH.py]: I now have 3 separate commits

**Pull request merged**